doc/design/model_zoo_dev.md

```diff
@@ -0,0 +1,207 @@
+# Design Doc: How to Define Deep Learning Models for SQLFlow
+
+## Configure Development Environment
+
+1. Install [Docker Community Edition](https://docs.docker.com/install/)
+
+2. Install SQLFlow as a Python package
+
+   ```bash
+   pip install sqlflow
+   ```
+
+3. Start SQLFlow server as a container running locally
+
+   ```bash
+   docker run --rm -it \
+     -p 3306:3306 \
+     -p 8888:8888 \
+     -e SQLFLOW_MYSQL_HOST=0.0.0.0 \
+     sqlflow/sqlflow:latest \
+     bash /start.sh
+   ```
+
+   The above command exposes MySQL server port 3306 out from the
+   SQLFlow server container, so we can use Navicat or VSCode MySQL
+   extension to manage data in the container.  It also exposes the
+   Jupyter Notebook port 8888, so users can run SQLFlow programs in
+   Jupyter Web page.
+
+Companies including Ant Financial and DiDi deployed SQLFlow and
+integrated into the corp data infrastructure.  Developers in these
+companies don't need to install any software, but can use the Web GUI
+of their data infrastructure.
+
+## Model Development in Python
+
+### Define a Model
+
+For clarify, let us consider a deep learning model definition is a
+Keras class in Python.  For example, the following class in
+`my_keras_model.py`:
+
+```python
+import sqlflow
+
+class MyKerasModel(tf.keras.Model):
+   def __init__(self, ...):
+      ...
+```
+
+For detailed instructions, please refer to this
+[document](https://github.com/sql-machine-learning/models/blob/develop/doc/customized%2Bmodel.md).
+
+### Test and Verify
+
+To verify that this model deifniton works, we can call Keras
+high-level API like `fit` to train a model and then evaluate it.
+
+After that, we want to verify it works with SQLFlow.  This depends on
+the SQLFlow high-level Python API, which is a Python abstraction of
+the SQL syntax extension introduced by SQLFlow.
+
+### SQLFlow High-level API
+
+For example, in SQLFlow we can write
+
+```sql
+SELECT ... TO TRAIN model_def WITH ... COLUMNS ... INTO trained_model;
+```
+
+Accordingly, in Python, we can write
+
+```python
+import sqlflow
+
+sqlflow.train(
```

Comment:
The general Python API should be like `sqlflow.execute("SELECT ... TO TRAIN model_def ... ", model_class= MyKerasModel)`, and the parameter `model_class` is optional, adding `model_class` so that the SQLFlow Python client can get the code of the model class and upload the code to SQLFlow server.

Reply:
为什么应该这样呢？是因为希望 sqlflow.train 只是构造一个 SQL statement 并且发给 SQLFlow server 吗？

Reply:
希望sqlflow 的python 客户端只保留最简单的功能：执行SQL语句。为了能让用户调试本地定义的模型程序，需要能将本地的模型Python代码发送到服务器端。所以会有参数`model_class= MyKerasModel`。